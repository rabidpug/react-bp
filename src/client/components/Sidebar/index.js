import {
  Icon,
  Layout,
  Menu,
} from 'antd';

import React from 'react';

const { Item, } = Menu;
const { Sider, } = Layout;
const Sidebar = ( {
  isSidebarCollapsed, menuItems, goToPath, toggleSideBar, router,
} ) => {
  const [ currentPath, ] = menuItems.filter( item => item.path === router.location.pathname );

  return (
    <Sider
      collapsed={ isSidebarCollapsed }
      collapsible
      onMouseEnter={ () => isSidebarCollapsed && toggleSideBar() }
      onMouseLeave={ () => !isSidebarCollapsed && toggleSideBar() }
      trigger={ null }>
      <div className='logo' />
      <Menu
        defaultSelectedKeys={ [ currentPath ? currentPath.key.toString() : '1', ] }
        mode='inline'
        onClick={ ( { item, } ) => goToPath( item.props.path ) }
        style={ { height: '100%', } }
        theme='dark'>
        {menuItems.map( item => (
          <Item
            key={ item.key }
            path={ item.path }>
            <Icon type={ item.icon } />
            <span>{item.label}</span>
          </Item>
        ) )}
      </Menu>
    </Sider>
  );
};

export default Sidebar;
