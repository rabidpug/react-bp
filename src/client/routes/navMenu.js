import { Home, UserForm, UserProfile, Welcome, } from 'Components/Loadables';
import { faComments, faHome, faSignInAlt, faUser, } from '@fortawesome/free-solid-svg-icons';

import SideBar from 'Containers/SideBar';

const navMenu = [
  {
    component : SideBar,
    menuItems : [
      {
        component : Home,
        icon      : faHome,
        key       : 'sub1',
        label     : 'Home',
        path      : '/',
        subMenu   : [
          {
            component : Home,
            icon      : faHome,
            key       : 'subitem1',
            label     : 'Home Page',
            path      : '/',
          },
        ],
      },
      {
        component : Welcome,
        icon      : faComments,
        key       : 'subitem2',
        label     : 'Welcome',
        path      : '/welcome',
      },
      {
        component       : UserForm,
        icon            : faSignInAlt,
        isAuthenticated : false,
        key             : 'item2',
        label           : 'Sign In/Up',
        path            : '/signin',
      },
      {
        component       : UserProfile,
        icon            : faUser,
        isAuthenticated : true,
        key             : 'item4',
        label           : 'User Profile',
        path            : '/profile',
      },
    ],
    path: '*',
  },
];

export default navMenu;
