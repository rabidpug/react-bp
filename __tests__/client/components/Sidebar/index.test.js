import React from 'react';
import Sidebar from '../../../../src/client/components/Sidebar';
import navMenu from '../../../../src/client/routes/navMenu';
import { shallow, } from 'enzyme';

describe(
  'Sidebar', () => {
    it(
      'Should render the Sidebar', () => {
        const [ { menuItems, }, ] = navMenu;
        const wrapper = shallow( <Sidebar
          menuItems={ menuItems }
          openKeys={ [] } /> );

        expect( wrapper ).toMatchSnapshot();
      }
    );
  }
);
