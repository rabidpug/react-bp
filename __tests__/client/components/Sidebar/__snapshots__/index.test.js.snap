// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should render the Sidebar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    menuItems={
      Array [
        Object {
          "icon": "home",
          "key": "sub1",
          "label": "Home",
          "path": "/",
          "subMenu": Array [
            Object {
              "icon": "home",
              "key": "subitem1",
              "label": "Home Page",
              "path": "/",
            },
          ],
        },
        Object {
          "icon": "smile-o",
          "key": "item1",
          "label": "Welcome",
          "path": "/welcome",
        },
        Object {
          "icon": "frown-o",
          "key": "item2",
          "label": "Bye",
          "path": "/bye",
        },
        Object {
          "icon": "file-text",
          "key": "item3",
          "label": "Todo",
          "path": "/todo",
        },
      ]
    }
    openKeys={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="logo"
        />,
        <Menu
          className=""
          mode="inline"
          onClick={[Function]}
          openKeys={Array []}
          prefixCls="ant-menu"
          selectedKeys={Array []}
          style={
            Object {
              "height": "100%",
            }
          }
          theme="dark"
        >
          <SubMenu
            onTitleClick={[Function]}
            title={
              <span>
                <Icon
                  type="home"
                />
                <span>
                  Home
                </span>
              </span>
            }
          >
            <MenuItem
              path="/"
            >
              <Icon
                type="home"
              />
              <span>
                Home Page
              </span>
            </MenuItem>
          </SubMenu>
          <MenuItem
            path="/welcome"
          >
            <Icon
              type="smile-o"
            />
            <span>
              Welcome
            </span>
          </MenuItem>
          <MenuItem
            path="/bye"
          >
            <Icon
              type="frown-o"
            />
            <span>
              Bye
            </span>
          </MenuItem>
          <MenuItem
            path="/todo"
          >
            <Icon
              type="file-text"
            />
            <span>
              Todo
            </span>
          </MenuItem>
        </Menu>,
      ],
      "collapsed": undefined,
      "collapsedWidth": 80,
      "collapsible": true,
      "defaultCollapsed": false,
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "prefixCls": "ant-layout-sider",
      "reverseArrow": false,
      "style": Object {
        "backgroundColor": "#fff",
      },
      "trigger": null,
      "width": 200,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "logo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <SubMenu
              onTitleClick={[Function]}
              title={
                <span>
                  <Icon
                    type="home"
                  />
                  <span>
                    Home
                  </span>
                </span>
              }
            >
              <MenuItem
                path="/"
              >
                <Icon
                  type="home"
                />
                <span>
                  Home Page
                </span>
              </MenuItem>
            </SubMenu>,
            <MenuItem
              path="/welcome"
            >
              <Icon
                type="smile-o"
              />
              <span>
                Welcome
              </span>
            </MenuItem>,
            <MenuItem
              path="/bye"
            >
              <Icon
                type="frown-o"
              />
              <span>
                Bye
              </span>
            </MenuItem>,
            <MenuItem
              path="/todo"
            >
              <Icon
                type="file-text"
              />
              <span>
                Todo
              </span>
            </MenuItem>,
          ],
          "className": "",
          "mode": "inline",
          "onClick": [Function],
          "openKeys": Array [],
          "prefixCls": "ant-menu",
          "selectedKeys": Array [],
          "style": Object {
            "height": "100%",
          },
          "theme": "dark",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "sub1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <MenuItem
                  path="/"
                >
                  <Icon
                    type="home"
                  />
                  <span>
                    Home Page
                  </span>
                </MenuItem>,
              ],
              "onTitleClick": [Function],
              "title": <span>
                <Icon
                  type="home"
                />
                <span>
                  Home
                </span>
              </span>,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "subitem1",
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Icon
                      type="home"
                    />,
                    <span>
                      Home Page
                    </span>,
                  ],
                  "path": "/",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "type": "home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Home Page",
                    },
                    "ref": null,
                    "rendered": "Home Page",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "item1",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  type="smile-o"
                />,
                <span>
                  Welcome
                </span>,
              ],
              "path": "/welcome",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "smile-o",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome",
                },
                "ref": null,
                "rendered": "Welcome",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "item2",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  type="frown-o"
                />,
                <span>
                  Bye
                </span>,
              ],
              "path": "/bye",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "frown-o",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Bye",
                },
                "ref": null,
                "rendered": "Bye",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "item3",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  type="file-text"
                />,
                <span>
                  Todo
                </span>,
              ],
              "path": "/todo",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "type": "file-text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Todo",
                },
                "ref": null,
                "rendered": "Todo",
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="logo"
          />,
          <Menu
            className=""
            mode="inline"
            onClick={[Function]}
            openKeys={Array []}
            prefixCls="ant-menu"
            selectedKeys={Array []}
            style={
              Object {
                "height": "100%",
              }
            }
            theme="dark"
          >
            <SubMenu
              onTitleClick={[Function]}
              title={
                <span>
                  <Icon
                    type="home"
                  />
                  <span>
                    Home
                  </span>
                </span>
              }
            >
              <MenuItem
                path="/"
              >
                <Icon
                  type="home"
                />
                <span>
                  Home Page
                </span>
              </MenuItem>
            </SubMenu>
            <MenuItem
              path="/welcome"
            >
              <Icon
                type="smile-o"
              />
              <span>
                Welcome
              </span>
            </MenuItem>
            <MenuItem
              path="/bye"
            >
              <Icon
                type="frown-o"
              />
              <span>
                Bye
              </span>
            </MenuItem>
            <MenuItem
              path="/todo"
            >
              <Icon
                type="file-text"
              />
              <span>
                Todo
              </span>
            </MenuItem>
          </Menu>,
        ],
        "collapsed": undefined,
        "collapsedWidth": 80,
        "collapsible": true,
        "defaultCollapsed": false,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "prefixCls": "ant-layout-sider",
        "reverseArrow": false,
        "style": Object {
          "backgroundColor": "#fff",
        },
        "trigger": null,
        "width": 200,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <SubMenu
                onTitleClick={[Function]}
                title={
                  <span>
                    <Icon
                      type="home"
                    />
                    <span>
                      Home
                    </span>
                  </span>
                }
              >
                <MenuItem
                  path="/"
                >
                  <Icon
                    type="home"
                  />
                  <span>
                    Home Page
                  </span>
                </MenuItem>
              </SubMenu>,
              <MenuItem
                path="/welcome"
              >
                <Icon
                  type="smile-o"
                />
                <span>
                  Welcome
                </span>
              </MenuItem>,
              <MenuItem
                path="/bye"
              >
                <Icon
                  type="frown-o"
                />
                <span>
                  Bye
                </span>
              </MenuItem>,
              <MenuItem
                path="/todo"
              >
                <Icon
                  type="file-text"
                />
                <span>
                  Todo
                </span>
              </MenuItem>,
            ],
            "className": "",
            "mode": "inline",
            "onClick": [Function],
            "openKeys": Array [],
            "prefixCls": "ant-menu",
            "selectedKeys": Array [],
            "style": Object {
              "height": "100%",
            },
            "theme": "dark",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "sub1",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <MenuItem
                    path="/"
                  >
                    <Icon
                      type="home"
                    />
                    <span>
                      Home Page
                    </span>
                  </MenuItem>,
                ],
                "onTitleClick": [Function],
                "title": <span>
                  <Icon
                    type="home"
                  />
                  <span>
                    Home
                  </span>
                </span>,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "subitem1",
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Icon
                        type="home"
                      />,
                      <span>
                        Home Page
                      </span>,
                    ],
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "type": "home",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Home Page",
                      },
                      "ref": null,
                      "rendered": "Home Page",
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "item1",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    type="smile-o"
                  />,
                  <span>
                    Welcome
                  </span>,
                ],
                "path": "/welcome",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "type": "smile-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome",
                  },
                  "ref": null,
                  "rendered": "Welcome",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "item2",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    type="frown-o"
                  />,
                  <span>
                    Bye
                  </span>,
                ],
                "path": "/bye",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "type": "frown-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Bye",
                  },
                  "ref": null,
                  "rendered": "Bye",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "item3",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    type="file-text"
                  />,
                  <span>
                    Todo
                  </span>,
                ],
                "path": "/todo",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "type": "file-text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Todo",
                  },
                  "ref": null,
                  "rendered": "Todo",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
